### PR TITLE
Complete feature Search

### PR DESCRIPTION
**Pull request by manhhnee**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/73271)

## WHAT (optional)
- Create feature search
## HOW
I create logic search for user and customer

## Evidence (Screenshot or Video)
image
image
image

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -2,4 +2,14 @@
 @import "style/auth/index";
 @import "style/home/index";
 @import "style/main";
-@import "style/admin/index";
+@import "header/index";
+@import "style/fields/index";
+@import "font-awesome";
+
+.message {
```
Comment (nguyentrieutien1) [inline app/assets/stylesheets/application.scss:9]: create new a css file to contain this messahe

**Review by nguyentrieutien1 — COMMENTED** [review]
Commented

**Comment by manhhnee** [comment]
ready

**Comment by manhhnee** [comment]
ready

File: app/controllers/static_pages_controller.rb [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
+    if params[:search].present?
+      @fields = Field.where("name LIKE ?", "%#{params[:search]}%")
+      if @fields.present?
+        @pagy, @fields = pagy @fields, items: Settings.PERPAGE_8
+      else
+        flash.now[:alert] = "Không tìm thấy kết quả phù hợp"
+      end
+    else
+      @pagy, @fields = pagy Field.sorted_by_name, items: Settings.PERPAGE_8
+    end
```
Comment (binhpt-1239) [inline app/controllers/static_pages_controller.rb:14]: Em viết 1 scope search_by_name nha

Trong đó
return nếu như value_search là blank?
Còn không thì thực hiện where

------------------------------
Ở controller em chỉ cần 
@pagy, @fields = pagy @fields.search_by_name(params[:search]), items: Settings.PERPAGE_8

**Comment by manhhnee** [comment]
ready

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 class StaticPagesController < ApplicationController
   def home
-    @pagy, @fields = pagy Field.sorted_by_name, items: Settings.PERPAGE_8
+    @fields = Field.search_by_name(params[:search]).sorted_by_name
+    @pagy, @fields = pagy @fields, items: Settings.PERPAGE_8
+    return unless @fields.blank?
+
+    flash.now[:alert] = t("controller.static_pages.not_found_field")
+
```
Comment (binhpt-1239) [inline app/controllers/static_pages_controller.rb:10]: Dư line

**Review by binhpt-1239 — APPROVED** [review]

**Pull request merged**